Components/ProductQuery/RemoteProductQuery.php

```diff
@@ -109,6 +124,15 @@ protected function getConnectProduct($row)
         $product = new Product(
             $row
         );
+
+        $this->eventManager->notify(
+            'Connect_Supplier_Get_Single_Product_Before',
```

Comment:
Do we need to use same name in LocalProductQuery and RemoteProductQuery, Connect_Supplier_Get_Single_Product_Before? 
In LocalProductQuery we fetch product which will be export from our shop. So in this case the shop is supplier.
In RemoteProductQuery we fetch product which is imported from supplier. So in this case the shop is merchant.